spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/context/MessageSourceAutoConfiguration.java

```
@@ -81,6 +89,25 @@ public MessageSource messageSource(MessageSourceProperties properties) {
 		}
 		messageSource.setAlwaysUseMessageFormat(properties.isAlwaysUseMessageFormat());
 		messageSource.setUseCodeAsDefaultMessage(properties.isUseCodeAsDefaultMessage());
+
+		try {
+			if (StringUtils.hasText(properties.getCommonMessages())) {
+				PropertiesFactoryBean propertiesFactory = new PropertiesFactoryBean();
+				ResourceLoader resourceLoader = new DefaultResourceLoader();
+				String[] commonMessages = StringUtils.commaDelimitedListToStringArray(
+						StringUtils.trimAllWhitespace(properties.getCommonMessages()));
+				List<Resource> commonResources = Arrays.stream(commonMessages)
+						.map(resourceLoader::getResource)
+						.toList();
+				propertiesFactory.setLocations(commonResources.toArray(Resource[]::new));
+				propertiesFactory.setSingleton(true);
+				propertiesFactory.setIgnoreResourceNotFound(true);
```

I think we should fail hard if the resource is not found.